### DISCUSSION TITLE
Delete sprite or not ?

### DISCUSSION THREAD

**Discussion by sbonaime** [Bodmer/TFT_eSPI#1943] [Q&A - Sprites] [answered]
Hi,
I am making something like this weather display
![68747470733a2f2f692e696d6775722e636f6d2f4f526f76774e592e706e67](https://user-images.githubusercontent.com/750709/182169728-c9b6cfd8-b8a8-444f-83bb-28c850ac3e40.png)

For each part of the display, I  use one sprite. 
What is the "best way" to deal with different sprites ? Do I have to createSprite and deleteSprite each time I draw each one ? Or can I use a "big" sprite to fit all the little parts in it ? Or Can I keep all my sprites unclosed  ? I know that a sprite needs (lots of) memory. Do createSprite and deleteSprite take cpu time ? 

Thanks

**Comment by Bodmer** [comment, marked as answer]
The answer depends on the detail in your implementation. If you have lots of RAM available then a full screen Sprite may be possible. It is also possible to re-use a sprite for different areas of the screen. If you are near to memory limits then creating and deleting sprites repeatedly can lead to memory fragmentation.
